pkg/util/kubernetes/apiserver/leaderelection/leaderelection_engine.go

```diff
@@ -219,14 +220,18 @@ func (le *LeaderEngine) reportLeaderMetric(isLeader bool) {
 	le.leaderMetric.Set(1.0, metrics.JoinLeaderValue, strconv.FormatBool(isLeader))
 }
 
-// notify sends a notification to subscribers when the current process becomes leader.
-// notify is a simplistic notifier but can be extended to send different notification
-// kinds (leadership acquisition / loss) in the future if needed.
+// notify sends a notification to subscribers when the leadership state of the current
+// process changes
 func (le *LeaderEngine) notify() {
 	le.m.Lock()
 	defer le.m.Unlock()
 
 	for _, s := range le.subscribers {
+		if len(s) > 0 {
```

Comment:
We should perhaps add a comment saying this code is racy, BUT there is now way to _loose_ a state as the as if the reader reads _after_ `len(s) > 0` check, it's still going to use the `isLeaderFunc` and retrieve the last **accurate** state.